src/components/annotations/draw.js

```diff
@@ -320,10 +341,18 @@ function drawRaw(gd, options, index, subplotId, xa, ya) {
             } else {
                 if(axLetter === 'x') {
                     alignPosition = options[axLetter];
-                    basePx = gs.l + gs.w * alignPosition;
+                    if(axRefType === 'domain') {
+                        basePx = ax._offset + ax._length * alignPosition;
+                    } else {
+                        basePx = gs.l + gs.w * alignPosition;
+                    }
```

Comment:
Let's store and reuse 
```js
var hasDomain = axRefType === 'domain';
```
on top.
Then you could avoid if statements for the setter e.g.
```js
basePx = hasDomain ?
  ax._offset + ax._length * alignPosition :
  gs.l + gs.w * alignPosition;
```